syntax = "proto3";
package user;

import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";


// 用户 ID
message UserId {
    // 用户 UUID
    string id = 1;
}


// 用户简要信息 用于展示
message UserShortInfo {
    // 用户 UUID
    string id = 1;
    // 用户名
    string username = 2;
    // 头像
    string avatar = 3;
}

// 用户详细信息
message UserInfo {
    // 用户 UUID
    string id = 1;
    // 用户名
    string username = 2;
    // 手机号
    string phone = 3;
    // 头像
    string avatar = 4;
    // 创建时间
    google.protobuf.Timestamp created_at = 5;
    // 更新时间
    google.protobuf.Timestamp updated_at = 6;
}

// 创建用户请求
message CreateUserRequest {
    // 手机号
    string phone = 1;
    // 验证码
    string code = 2;
    // 用户名
    string username = 3;
    // 密码
    string password = 4;
}

// 更新用户头像请求
message UpdateUserAvatarRequest {
    // 用户 UUID
    string id = 1;
    // 头像
    string avatar = 2;
}

message UpdateUserInfoRequest {
    // 用户 UUID
    string id = 1;
    // 用户名
    string username = 2;
    // 旧密码
    string old_password = 3;
    // 新密码
    string new_password = 4;
    // 手机号
    string phone = 5;
    // 验证码
    string code = 6;
}


// 用户服务接口
service User {
    // 获取指定ID用户简要信息
    rpc GetUserShortInfo (UserId) returns (UserShortInfo);
    // 获取指定ID用户详细信息
    rpc GetUserInfo (UserId) returns (UserInfo);

    // 更新用户头像
    rpc UpdateUserAvatar (UpdateUserAvatarRequest) returns (UserInfo);
}

// 用户公开接口
service UserPublic {
    // 获取指定ID用户简要信息
    rpc GetUserShortInfo (UserId) returns (UserShortInfo);

    // 获取当前用户信息
    rpc GetCurrentUserInfo (google.protobuf.Empty) returns (UserInfo);

    // 更新用户信息
    rpc UpdateUserInfo (UpdateUserInfoRequest) returns (UserInfo);
}
